enterprise/views.py

```diff
@@ -468,6 +482,24 @@ def get_enterprise_course_enrollment_page(self, request, enterprise_uuid, course
 
         """
         try:
+            enterprise_uuid = UUID(enterprise_uuid)
+            # pylint: disable=no-member
+            enterprise_customer = EnterpriseCustomer.objects.get(uuid=enterprise_uuid)
+        except (ValueError, EnterpriseCustomer.DoesNotExist):
+            logger.error('Unable to find enterprise customer for UUID: %s', enterprise_uuid)
+            raise Http404
+
+        # Now verify if the user is logged in. If user is not logged in then
+        # send the user to the login screen to sign in with an
+        # Enterprise-linked SSO and the pipeline will get them back here.
+        if not request.user.is_authenticated():
+            next_url = '{current_url}?{query_string}'.format(
+                current_url=quote(request.get_full_path()),
+                query_string=urlencode({'tpa_hint': enterprise_customer.identity_provider})
+            )
+            return redirect('{login_url}?next={next_url}'.format(login_url='/login', next_url=next_url))
```

Comment:
Does the `tpa_hint` here prevent us from using the `@login_required` decorator? If so, can we create a new decorator that implements this logic?